features/step_definitions/git-flow.js

```diff
@@ -104,90 +16,44 @@ function promiseToWait(timeToWait) {
     });
 }
 
-/**
- * Search for a build that is executing against a specific SHA
- * @method searchForBuild
- * @param  {String}       screwdriverInstance  Specific screwdriver instance to query against
- * @param  {String}       desiredSha           The SHA that the build is using
- * @param  {Number}       [pageNumber]         Some page number
- * @return {Promise}
- */
-function searchForBuild(screwdriverInstance, desiredSha, pageNumber) {
-    const pageCounter = pageNumber || 1;
-
-    return request({
-        json: true,
-        method: 'GET',
-        uri: `${screwdriverInstance}/v4/builds?page=${pageCounter}&count=${MAX_PAGE_COUNT}`
-    })
-    .then((response) => {
-        const buildData = response.body;
-        const result = buildData.filter((build) => build.sha === desiredSha);
-
-        if (buildData.length === MAX_PAGE_COUNT) {
-            return searchForBuild(screwdriverInstance, desiredSha, pageCounter + 1)
-            .then((nextPage) => result.concat(nextPage));
-        }
-
-        return result;
-    });
-}
-
-/**
- * Perisistently ping the API until the build data is available
- * @method waitForBuild
- * @param  {String}     screwdriverInstance  Specific Screwdriver instance to query against
- * @param  {String}       desiredSha         The SHA that the build is using
- * @return {Promise}
- */
-function waitForBuild(screwdriverInstance, desiredSha) {
-    console.log('    (Waiting for build to exist....)');
-
-    return searchForBuild(screwdriverInstance, desiredSha)
-    .then((buildData) => {
-        if (buildData.length !== 0) {
-            return buildData;
-        }
-
-        return promiseToWait(3)
-            .then(() => searchForBuild(screwdriverInstance, desiredSha));
-    });
-}
-
 module.exports = function server() {
     // eslint-disable-next-line new-cap
-    this.Before(() => {
+    this.Before({
+        tags: ['@gitflow'],
+        timeout: 60000
+    }, () => {
         this.instance = 'https://api.screwdriver.cd';
-        this.repositoryOwner = 'screwdriver-cd';
-        this.repository = 'garbage-repository-ignore-this';
-        this.testBranch = 'testBranch';
-        const params = {
-            user: this.repositoryOwner,
-            repo: this.repository,
-            ref: `heads/${this.testBranch}`
-        };
+        this.branch = 'darrenBranch';
+        this.repoOrg = 'screwdriver-cd-test';
+        this.repoName = 'functional-git';
+        this.pipelineId = '2e0138dfa7c4ff83720dc0cd510d2252a3398fc3';  // TODO: determine dynamically
```

Comment:
Nice to know you can set up the Before hook for a given tag! But we already have `this.instance` in the before-hook under support. Do we really need it here?